src/HypothesisClient/Credentials/JWTSigningCredentials.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace eLife\HypothesisClient\Credentials;
+
+use Firebase\JWT\JWT;
+
+class JWTSigningCredentials extends Credentials
+{
+    private $expire;
+    private $startTime;
+
+    public function __construct(string $clientId, string $clientSecret, string $authority, int $expire = 600, int $startTime = null)
```

Comment:
was looking for where `JWTSigningCredentials` is instantiated in the Kernel, but found only the tests
would make `startTime` mandatory rather than default, to force creation code to pass it and express the dependency and non-determinism (moving it up in the list then as `$expire` seems to have a safe default)

Reply:
I will be introducing it to the Kernel later. These are necessary refactors to pave the way for the annotations API.

Reply:
I intend for `JWTSigningCredntials` to be instantiated in `Kernel` and so setting a `startTime` as a constructor argument is not going to work when we need to make token requests in the long running process. We really need the start time to be reset every when we generate the JWT.

Reply:
If a different time value is required each time, a common pattern a is to inject a `Clock` object instead of the integer. A lightweight version of that is to inject a function returning an integer:
`__construct(..., callable $time)`
Then, the class can be instantiated like this:
```
new JWTSigningCredentials(..., 'time'); // Kernel
new JWTSigningCredentials(..., function() { return '1500000000'; }); // tests
```

Reply:
I have added a `Clock` object.